### PR TITLE
Feature: Created Lookup and Routing

### PR DESCRIPTION
**Pull request by AGreen153**
- Created route between MusicArea.tsx, StepSequencer.tsx, and BrowseSongs.tsx
- Created BrowseSongs with temporary variables (will be fetched from database in the future)
- Search feature

### CODE REVIEW

File: frontend/src/components/Town/interactables/Music/StepSequencer.tsx [hunk 1]
```diff
@@ -80,6 +157,8 @@ export default function MusicArea({
         if (sound.playNote) {
           console.log('Playing Sound: ' + sound.note);
           if (sound.note == 'Cymbol' || sound.note == 'Drum') {
+            // Was synth.triggerAttackRelease('16n', time); but ran into this error in console.log() when spamming
```
Comment (Ironcrabman) [inline frontend/src/components/Town/interactables/Music/StepSequencer.tsx:167]: Can we get this listed as an official bug on the issues page?

File: frontend/src/components/Town/interactables/Music/StepSequencer.tsx [hunk 2]
```diff
@@ -168,8 +248,142 @@ export default function MusicArea({
...
+    if (event.key === 'Enter') {
+      console.log('TODO: Search the database for the title!');
+    } else {
+      coveyTownController.pause();
+    }
+  };
+
+  /* Whenever the user releases a button, unpause the game*/
+  const handleKeyUp = () => {
+    // TODO: This is broken, there's some logic here that's wrong (compare it to the basement dining table below and the code at /components/Town/interactables/NewConversationModal.tsx)
```
Comment (Ironcrabman) [inline frontend/src/components/Town/interactables/Music/StepSequencer.tsx]: Nit: multiple Todo's are included throughout this pr, is it possible to list those at the top of each file in order to keep track of which functionalities are not fully implemented yet. 
EX. Saving the song information after the save checks

File: frontend/src/components/Town/interactables/Music/BrowseSongs.tsx [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+}): JSX.Element {
+  const [songs, setSongs] = useState<Song[]>([]);
+  const coveyTownController = useTownController();
+
+  /* TODO: At load this should fetch info from the backend, and load setSongs for now it's just a temporary array of fake data */
+  useEffect(() => {
+    /* TODO: Make database call to fetch the songs and load it into setSongs instead of the fakeSongs useEffect below*/
+  }, []);
+
+  /* TODO: Delete this, this just loads fake data into songs, used for styling purposes */
```
Comment (M3CHR0M4NC3R) [inline frontend/src/components/Town/interactables/Music/BrowseSongs.tsx:25]: will be resolved by [Music-Storage](https://github.com/M3CHR0M4NC3R/covey.town/tree/Music-Storage)
These fake songs are going into their own files and will be read in

File: frontend/src/components/Town/interactables/Music/StepSequencer.tsx [hunk 4]
```diff
@@ -168,8 +248,142 @@ export default function MusicArea({
...
+    // Save the song to the database
+
+    // Update the user
+    setError('Song Saved!');
+  };
+
+  // TODO: Like the song to the database (pass it the board object)
+  const likeSong = () => {
+    // Check if the user has already liked the song
+    const username = '';
```
Comment (Ironcrabman) [inline frontend/src/components/Town/interactables/Music/StepSequencer.tsx:326]: Major: Still need to implement the method for getting the username. Should be marked as todo and listed at the top of file

File: frontend/src/components/Town/interactables/Music/StepSequencer.tsx (hunk 1 above)
Comment (AGreen153) [reply frontend/src/components/Town/interactables/Music/StepSequencer.tsx:167]: Done!

File: frontend/src/components/Town/interactables/Music/StepSequencer.tsx (hunk 2 above)
Comment (AGreen153) [reply frontend/src/components/Town/interactables/Music/StepSequencer.tsx]: Done!

**Pull request merged**